reporter/html.js

```diff
@@ -370,15 +374,27 @@ export function escapeText( s ) {
 		label.innerHTML = "Module: ";
 		label.appendChild( moduleSearch );
 
+		applyButton.textContent = "Apply";
+		applyButton.style.display = "none";
```

Comment:
I haven't confirmed, so apologies if this isn't true... I would expect the CSP policy for inline styles to disallow all three methods of dynamic style creation: `<style>` elements, `style=""` attributes, and CSSOM/style properties (like this one).

To abide the CSP strict style policy, I think we need to declare the styles in the CSS file ahead of time, and toggle them with a class only. For example, the `qunit-modulefilter-dropdown` element could have a class like `has-selection` or `has-no-selection`. And based on that, we could hide the buttons as needed.

However, the apply button isn't the only violation, so to address your use case we'll need numerous other changes as well. Again, that's based on my (potentially) outdated understanding of CSP.

Reply:
The specification is [not quite clear regarding CSSOM and CSP](https://www.w3.org/TR/CSP3/#directive-style-src) but as far as I'm aware of none of the implementation considers manipulating styles through CSSOM as a CSP violation. So while `document.create('<button style="display: none;">')` and `document.create('button').style = "display: none;"` require `style-src 'unsafe-inline'`, `document.create('button').style.display = 'none'` is not violation `style-src: 'none'`.